ibm/service/cis/resource_ibm_cis_mtls_app.go

```diff
@@ -0,0 +1,285 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package cis
+
+import (
+	"fmt"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/networking-go-sdk/mtlsv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	cisMtlsAppName       = "app_name"
+	cisMtlsDuration      = "duration"
+	cisMtlsURL           = "url"
+	cisMtlsRuleCertVal   = "rule_cert"
+	cisMtlsRuleCommonVal = "rule_common"
+	cisMtlsPolicyName    = "policy_name"
+	cisMtlsPolicyAction  = "policy_action"
+)
+
+func ResourceIBMCISMtlsApp() *schema.Resource {
+	return &schema.Resource{
+		Create:   ResourceIBMCISMtlsAppCreate,
+		Read:     ResourceIBMCISMtlsAppRead,
+		Update:   ResourceIBMCISMtlsAppUpdate,
+		Delete:   ResourceIBMCISMtlsAppDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisMtlsHostName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Host associated with",
+			},
+			cisMtlsAppName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "App Name",
+			},
+			cisMtlsDuration: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "24h",
+				Description: "Duration for app validatidity",
+			},
+			cisMtlsURL: {
```

Comment:
cisMtlsURL is this required ? we are not reading it Create method (its in update) shd it be optional ? If required then create shd call update

Reply:
It is redundant field removed now (hostname is required)